src/targets/gpu/convolution.cpp

```diff
@@ -55,14 +59,53 @@ argument miopen_convolution::compute(context& ctx,
                                      const shape& output_shape,
                                      const std::vector<argument>& args) const
 {
-    auto x_desc = make_tensor(reshape_if_1d(args[0].get_shape()));
-    auto w_desc = make_tensor(reshape_if_1d(args[1].get_shape()));
-    auto y_desc = make_tensor(reshape_if_1d(output_shape));
+    auto x_desc                = make_tensor(reshape_if_1d(args[0].get_shape()));
+    auto w_desc                = make_tensor(reshape_if_1d(args[1].get_shape()));
+    auto y_desc                = make_tensor(reshape_if_1d(output_shape));
+    auto* miopen_stream_handle = ctx.get_stream().get_miopen();
+    auto workspace_size        = args[2].get_shape().bytes();
+
+#ifdef MIGRAPHX_HAS_FIND_2_API
+    {
+        const miopenTensorArgumentId_t names[3] = {
+            miopenTensorConvolutionX, miopenTensorConvolutionW, miopenTensorConvolutionY};
+
+        // cppcheck-suppress constVariable
+        void* buffers[3] = {args[0].implicit(), args[1].implicit(), args[3].implicit()};
+
+        miopenTensorDescriptor_t descriptors[3] = {x_desc.get(), w_desc.get(), y_desc.get()};
+
+        auto arguments = std::make_unique<miopenTensorArgument_t[]>(3);
+
+        for(auto i = 0; i < 3; ++i)
+        {
+            arguments[i].id         = names[i];
+            arguments[i].descriptor = descriptors[i] != nullptr ? &descriptors[i] : nullptr;
+            arguments[i].buffer     = buffers[i];
+        }
```

Comment:
Is there way to retrieve descriptors from `miopenSolution`  object ? Ideally we would want to assert that they have not been changed.

Reply:
I could do that, but I think that it would be better design-wise to either make your object effectively immutable or mark it as dirty on changes, rather than fetch data every time to compare.

Reply:
Yes, i think it is unlikely case that descriptors would change. @pfultz2  Can you provide your thoughts ?

Reply:
And if they have changed, you would need a new find for almost any solution even in the future as it would need a new compilation or at least a new invoker.

Reply:
We can store the descriptors in the convolution object and create them in finalize if we want to, but I dont see a problem with creating them on the fly. I haven't really seen a perf difference with it.